scripts/zones/Batallia_Downs_[S]/npcs/Thorben.lua

```diff
@@ -0,0 +1,63 @@
+-----------------------------------
+-- Area: Batallia Downs (S)
+--  NPC: Thorben
+-- Type: Quest NPC
+-- !pos  175.346,8.038,-419.244 84
+-----------------------------------
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+require("scripts/globals/quests")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local lostInTranslocation = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.LOST_IN_TRANSLOCATION)
+    local leftMapPiece = player:hasKeyItem(dsp.ki.LEFT_MAP_PIECE)
+    local middleMapPiece = player:hasKeyItem(dsp.ki.MIDDLE_MAP_PIECE)
+    local rightMapPiece = player:hasKeyItem(dsp.ki.RIGHT_MAP_PIECE)
+    local anyMapPiece = leftMapPiece or middleMapPiece or rightMapPiece
+
+    if lostInTranslocation == QUEST_COMPLETED then
+        player:startEvent(108)
+    elseif lostInTranslocation == QUEST_ACCEPTED and leftMapPiece and middleMapPiece and rightMapPiece then
+        player:startEvent(107)
+    elseif player:getVar("lostInTranslocationCS") == 1 and anyMapPiece then
```

Comment:
One tiny last thing I noticed: if the var is set after 105, then we already know the player has a map piece and don't need to check again. :wink:

Reply:
good catch 😸️ made the change. Thank you